drivers/interrupt_controller/intc_ite_it8xxx2.c

```diff
@@ -69,6 +69,30 @@ inline void set_csr(unsigned long bit)
 	}
 }
 
+#define IT8XXX2_IER_COUNT ARRAY_SIZE(reg_enable)
+static uint8_t ier_setting[IT8XXX2_IER_COUNT];
+
+void ite_intc_save_and_disable_interrupts(void)
+{
+	/* Disable global interrupt for critical section */
+	unsigned int key = irq_lock();
+
+	/* Save and disable interrupts */
+	for (int i = 0; i < IT8XXX2_IER_COUNT; i++) {
+		ier_setting[i] = *reg_enable[i];
+		*reg_enable[i] = 0;
+	}
+	irq_unlock(key);
+}
+
+void ite_intc_restore_interrupts(void)
+{
+	/* Restore interrupt state */
```

Comment:
Should you irq_lock around this to make sure all are enabled to get the highest priority to take off first instead of the first one enabled that is ready to go?

Reply:
Done.